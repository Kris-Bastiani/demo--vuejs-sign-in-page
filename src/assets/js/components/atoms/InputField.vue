<template lang="pug">
	label.input_field(:class='{ _invalid: error }')
		span.input_field__label {{ label }}
		input(:placeholder='placeholder', :type='type', :value='value', @input="$emit('input', $event.target.value)")
		.input_field__error(v-if='error') {{ error }}
</template>

<script>
export default {
	props: {
		error: String,
		label: String,
		placeholder: String,
		type: String,
		value: String,
	},
};
</script>
