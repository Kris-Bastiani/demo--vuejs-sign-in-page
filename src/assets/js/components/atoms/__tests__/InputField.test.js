import { mount } from '@vue/test-utils';
import InputField, { INPUT_TYPES } from '../InputField.vue';

const mountInputField = propsData => mount(InputField, { propsData });

describe('InputField', () => {
	INPUT_TYPES.forEach(type => it(`Should match snapshot when type = ${type}`, () => {
		const wrapper = mountInputField({ label: type, type });
		expect(wrapper.element).toMatchSnapshot();
	}));

	it('Should match snapshot when an error message is provided', () => {
		const wrapper = mountInputField({
			error: 'This is an error message.',
			label: 'text',
			type: 'text',
		});
		expect(wrapper.element).toMatchSnapshot();
	});
});
