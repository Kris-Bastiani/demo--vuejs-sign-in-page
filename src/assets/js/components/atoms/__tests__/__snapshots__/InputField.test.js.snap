// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Should match snapshot when an error message is provided 1`] = `
<label
  class="input_field _invalid"
>
  <span
    class="input_field__label"
  >
    text
  </span>
  <input
    type="text"
  />
  <p
    class="message _error"
  >
    This is an error message.
  </p>
</label>
`;

exports[`InputField Should match snapshot when type = color 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    color
  </span>
  <input
    type="color"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = date 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    date
  </span>
  <input
    type="date"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = datetime-local 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    datetime-local
  </span>
  <input
    type="datetime-local"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = email 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    email
  </span>
  <input
    type="email"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = file 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    file
  </span>
  <input
    type="file"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = month 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    month
  </span>
  <input
    type="month"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = number 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    number
  </span>
  <input
    type="number"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = password 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    password
  </span>
  <input
    type="password"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = search 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    search
  </span>
  <input
    type="search"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = tel 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    tel
  </span>
  <input
    type="tel"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = text 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    text
  </span>
  <input
    type="text"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = time 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    time
  </span>
  <input
    type="time"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = url 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    url
  </span>
  <input
    type="url"
  />
  <!---->
</label>
`;

exports[`InputField Should match snapshot when type = week 1`] = `
<label
  class="input_field"
>
  <span
    class="input_field__label"
  >
    week
  </span>
  <input
    type="week"
  />
  <!---->
</label>
`;
